src/service/cia301/co_pdo.c

```diff
@@ -602,6 +606,12 @@ void CORPdoWrite(CO_RPDO *pdo, CO_IF_FRM *frm)
                     }
                     dlc += pdosz;
                     COObjWrValue(obj, pdo->Node, (void *)&val32, sz);
+                } else if (sz == 6u) {
+                    val64 = CO_GET_LONG(frm, dlc);
+                    val64 |= (uint64_t)(CO_GET_WORD(frm, dlc + 4)) << 32;
+                    val64 &= 0x0000FFFFFFFFFFFF;
```

Comment:
The line

```c
val64 &= 0x0000FFFFFFFFFFFF;
```

is not needed.

Reply:
I agree, it's only that I've imitated INTEGER24:
https://github.com/embedded-office/canopen-stack/blob/master/src/service/cia301/co_pdo.c#L601
then I can create a patch removing the:
```
val32 &= 0x00FFFFFF;
```
that doesn't make sense at all to me.
What about it?

Reply:
Yes, right. I have implemented the Integer24 not very smart. In parallel I have prepared a similar implementation for Integer24 inspired by your idea how to handle Integer48. In this commit I will remove the

```c
val &= 0x00FFFFFF;
```

as well ;-)

Reply:
This increases a few memory footprint but gives more consistency on object dictionary.
Your implementation was good, the only "problem" is that in Object dictionary we find TUNSIGNED32 where instead we have a UNSIGNED24.

What about instead if I implement UNSIGNED64 and then we simply:
```
#define CO_TUNSIGNED24    CO_TUNSIGNED32
#define CO_TUNSIGNED40    CO_TUNSIGNED64
#define CO_TUNSIGNED48    CO_TUNSIGNED64
#define CO_TUNSIGNED56    CO_TUNSIGNED64
```
?

And then we handle UNSIGNED24/40/48/56 the way you've done for INTEGER24 with:
```
val32 &= 0x00FFFFFF;
```
This decreases memory footprint and avoid to repeat code.
So I'd drop integer48.c and instead I'd add integer64.c

I can do all tests here for all data types and check it works correctly.